sdk/sqlmanagement/Azure.ResourceManager.Sql/tests/Scenario/ElasticPoolTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Azure.ResourceManager.Sql.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Sql.Tests.Scenario
+{
+    public class ElasticPoolTests : SqlManagementTestBase
+    {
+        private ResourceGroupResource _resourceGroup;
+        private ResourceIdentifier _resourceGroupIdentifier;
+        private static AzureLocation Location = new AzureLocation("uksouth", "UK South");
+
+        public ElasticPoolTests(bool isAsync)
+            : base(isAsync)//, RecordedTestMode.Record)
+        {
+        }
+
+        [OneTimeSetUp]
+        public async Task GlobalSetUp()
+        {
+            var rgLro = await GlobalClient.GetDefaultSubscriptionAsync().Result.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, SessionRecording.GenerateAssetName("Sql-RG-"), new ResourceGroupData(Location));
+            ResourceGroupResource rg = rgLro.Value;
+            _resourceGroupIdentifier = rg.Id;
+            await StopSessionRecordingAsync();
+        }
+
+        [SetUp]
+        public async Task TestSetUp()
+        {
+            ArmClientOptions options = new ArmClientOptions();
+            var client = GetArmClient(options);
+            _resourceGroup = await client.GetResourceGroupResource(_resourceGroupIdentifier).GetAsync();
+        }
+
+        [TearDown]
+        public async Task TearDown()
+        {
+            var sqlServerList = await _resourceGroup.GetSqlServers().GetAllAsync().ToEnumerableAsync();
+            foreach (var item in sqlServerList)
+            {
+                await item.DeleteAsync(WaitUntil.Completed);
+            }
+        }
+
+        [Test]
+        [RecordedTest]
+        public async Task ElasticPoolApiTests()
+        {
+            // create Sql Server
+            string serverName = Recording.GenerateAssetName("sql-server-");
```

Comment:
Consider putting the shared logic to setup, such as sql server creation and collection declaration.

Reply:
updated